mp/src/game/client/neo/ui/neo_hud_ghost_marker.cpp

```diff
@@ -111,31 +111,49 @@ void CNEOHud_GhostMarker::DrawNeoHudElement()
 
 	const float scale = neo_ghost_marker_hud_scale_factor.GetFloat();
 
-	const int offset_X = m_iPosX - ((m_iMarkerTexWidth * 0.5f) * scale);
-	const int offset_Y = m_iPosY - ((m_iMarkerTexHeight * 0.5f) * scale);
-
-	Color ghostColor = COLOR_GREY;
-	if (m_iGhostingTeam == TEAM_JINRAI || m_iGhostingTeam == TEAM_NSF)
-	{
-		if ((m_iClientTeam == TEAM_JINRAI || m_iClientTeam == TEAM_NSF) && (m_iClientTeam != m_iGhostingTeam))
+	// The increase and decrease in alpha over 6 seconds
+	float alpha6 = remainder(gpGlobals->curtime, 6) + 3;
+	alpha6 = alpha6 / 6;
+	if (alpha6 > 0.5)
+		alpha6 = 1 - alpha6;
+	alpha6 = 128 * alpha6;
+
+	for (int i = 0; i < 4; i++) {
+		m_fMarkerScalesCurrent[i] = (remainder(gpGlobals->curtime, 2) / 2) + 0.5 + m_fMarkerScalesStart[i];
+		if (m_fMarkerScalesCurrent[i] > 1)
+			m_fMarkerScalesCurrent[i] -= 1;
+
+		const int offset_X = m_iPosX - ((m_iMarkerTexWidth * 0.5f * m_fMarkerScalesCurrent[i]) * scale);
+		const int offset_Y = m_iPosY - ((m_iMarkerTexHeight * 0.5f * m_fMarkerScalesCurrent[i]) * scale);
+
+		int alpha = 64 + alpha6;
+		if (m_fMarkerScalesCurrent[i] > 0.5)
+			alpha *= ((0.5 - (m_fMarkerScalesCurrent[i] - 0.5)) * 2);
+
+		Color ghostColor = Color(204, 204, 204, alpha);
+		if (m_iGhostingTeam == TEAM_JINRAI || m_iGhostingTeam == TEAM_NSF)
 		{
-			// If viewing from playing player, but opposite of ghosting team, show red
-			ghostColor = COLOR_RED;
-		}
-		else
-		{
-			// Otherwise show ghosting team color (if friendly or spec)
-			ghostColor = (m_iGhostingTeam == TEAM_JINRAI) ? COLOR_JINRAI : COLOR_NSF;
+			if ((m_iClientTeam == TEAM_JINRAI || m_iClientTeam == TEAM_NSF) && (m_iClientTeam != m_iGhostingTeam))
+			{
+				// If viewing from playing player, but opposite of ghosting team, show red
+				ghostColor = COLOR_RED;
+			}
+			else
+			{
+				// Otherwise show ghosting team color (if friendly or spec)
+				ghostColor = (m_iGhostingTeam == TEAM_JINRAI) ? COLOR_JINRAI : COLOR_NSF;
+			}
 		}
```

Comment:
These will override the alpha values. Set the initial colors then edit the alpha value on the color.

Reply:
yeah thats why i marked it as a draft but thanks xd

Reply:
I was thinking more like `ghostColor[3] = alpha` so the colors can still be used but it's the same result.